server/src/testFixtures/java/io/crate/testing/SQLExecutor.java

```diff
@@ -907,6 +916,14 @@ public <T> T plan(String statement) {
         return plan(statement, UUID.randomUUID(), 0);
     }
 
+    public void updateTableStats(Map<RelationName, Stats> stats) {
+        tableStats.updateTableStats(stats);
+    }
```

Comment:
The Builder already has a `setTableStats(TableStats tableStats)`, I suspect we don't need both?

Reply:
This is used to update the stats not to set it. See usage here:

https://github.com/crate/crate/blob/7b84ef97b8de2f0e97ef2e34bc6111155042d6be/server/src/test/java/io/crate/planner/operators/JoinTest.java#LL128C1-L147C6

Reply:
`setTableStats` would require rebuilding the `SQLExecutor` to update. However, i think i can remove that.

Reply:
No it doesn't, you can set it when you build it initially:

```diff
     private PlannerContext plannerCtx;
     private final CoordinatorTxnCtx txnCtx = CoordinatorTxnCtx.systemTransactionContext();
+    private TableStats tableStats;

     @Before
     public void setUpExecutor() throws IOException {
+        tableStats = new TableStats();
         e = SQLExecutor.builder(clusterService, 2, Randomness.get(), List.of())
+            .setTableStats(tableStats)
             .addTable(USER_TABLE_DEFINITION)
             .addTable(TEST_DOC_LOCATIONS_TABLE_DEFINITION)
```

And then call `updateTableStats` in the test as needed